### PR TITLE
Feature/game logic

### PR DESCRIPTION
**Pull request by uryelah**
- Created game_logic.rb file containing the logic of the game
- Added classes, Game and Player, in the game_logic.rb
- Added single-responsibility methods to classes
- Added the validations to avoid cheating from the players

### CODE REVIEW

File: lib/game_logic.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def winner(vals)
+    return 'P1' if row_winner(vals, 'O')
+    return 'P2' if row_winner(vals, 'X')
+    return 'P1' if column_winner(vals, 'O')
+    return 'P2' if column_winner(vals, 'X')
+    return 'P1' if diagonal_winner(vals, 'O')
+    return 'P2' if diagonal_winner(vals, 'X')
+    return 'TIE' if check_tie(vals)
```
Comment (Ispirett) [inline lib/game_logic.rb]: Don't use hardcoded values to like 'X and 'O' here instead let's use the player object icon

File: lib/game_logic.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  def display_board
+    "    1  2  3\n a [#{@board[:a1]}][#{@board[:a2]}][#{@board[:a3]}]
+ b [#{@board[:b1]}][#{@board[:b2]}][#{@board[:b3]}]
+ c [#{@board[:c1]}][#{@board[:c2]}][#{@board[:c3]}]\n\n"
+  end
+
+  def board_values
+    @board.values.join
+  end
+
```
Comment (Ispirett) [inline lib/game_logic.rb]: **Optional**  It will be better to put board related methods in a `Board  class`

File: bin/main.rb [hunk 3]
```diff
@@ -1,3 +1,72 @@
...
+puts "INSTRUCTIONS:\n\n"
+puts "Tic-tac-toe is a very fun game!\n#{player1} gets the 'X' and #{player2} the 'O'\n\n"
+puts "The first one to connect three X's O's sequence is the Winner!\n\n"
+puts "If you want to quit the game, type 'quit' after you choose names\n\n"
+puts 'Select a cell by typing the row (letter) followed by the column (number)'
+puts "For example: 'a1'\n\n"
+
+puts "Here's your Tic-tac-toe game board!\n\n"
+
+puts tic_tac.display_board
```
Comment (Ispirett) [inline bin/main.rb]: **Optional** you can wrap some of the logic in this file into functions to make you code more readable.

**Review by Ispirett — APPROVED** [review]
## Status: approved :1st_place_medal: 

Good Job @uryelah @majovainilla  you did great work with this project :+1:   :1st_place_medal: 

Please remove the javascript and JSON files from the bin folder thanks.

File: lib/game_logic.rb (hunk 1 above)
Comment (uryelah) [reply lib/game_logic.rb]: Done, Amazing that it was that easy :-D

File: lib/game_logic.rb (hunk 2 above)
Comment (uryelah) [reply lib/game_logic.rb]: Done. Great suggestion!

File: bin/main.rb (hunk 3 above)
Comment (uryelah) [reply bin/main.rb]: Done :-D

**Pull request merged**